pkg/graph/schema.resolvers.go

```diff
@@ -1194,15 +1194,31 @@ func (r *queryResolver) SystemIntake(ctx context.Context, id uuid.UUID) (*models
 		return nil, err
 	}
 
-	ok, err := services.AuthorizeUserIsIntakeRequesterOrHasGRTJobCode(ctx, intake)
+	// if this user created the intake
+	if ok := services.AuthorizeUserIsIntakeRequester(ctx, intake); ok {
+		return intake, nil
+	}
+
+	// if this user is an admin
+	if ok := services.AuthorizeRequireGRTJobCode(ctx); ok {
+		return intake, nil
+	}
+
+	grbUsers, err := r.store.SystemIntakeGRBReviewersBySystemIntakeIDs(ctx, []uuid.UUID{id})
 	if err != nil {
 		return nil, err
 	}
-	if !ok {
-		return nil, &apperrors.UnauthorizedError{Err: errors.New("unauthorized to fetch system intake")}
+
+	principal := appcontext.Principal(ctx)
+
+	// if this user is a GRB viewer/voter for this intake
+	if slices.ContainsFunc(grbUsers, func(reviewer *models.SystemIntakeGRBReviewer) bool {
+		return reviewer.UserID == principal.Account().ID
+	}) {
+		return intake, nil
```

Comment:
You could avoid the comment by making a variable, but it's also neat you can just drop the fn call directly into the if statement:

```suggestion
	isGRBReviewer := slices.ContainsFunc(
	    grbUsers, 
	    func(reviewer *models.SystemIntakeGRBReviewer) bool {
		return reviewer.UserID == principal.Account().ID
	    }
	)
	if isGRBReviewer {
		return intake, nil
```

Reply:
that's cleaner - updating. thanks!